drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -117,6 +152,10 @@ static void adc_context_start_sampling(struct adc_context *ctx)
 	/* Extract lower channel from sequence mask */
 	int current_channel = u32_count_trailing_zeros(data->channel_read_mask);
 
+	GPADC->GP_ADC_CTRL3_REG = 0x40;
+	GPADC->GP_ADC_CTRL_REG = GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk;
+	GPADC->GP_ADC_CTRL2_REG = 0;
```

Comment:
No need, to explicitly zero that register as it will be written right below:
`GPADC->GP_ADC_CTRL2_REG = m_channels[current_channel].gp_adc_ctrl2_reg;`